kong/plugins/pre-function/_handler.lua

```diff
@@ -11,37 +9,82 @@ return function(plugin_name, priority)
     VERSION = "0.3.1",
   }
 
-  function ServerlessFunction:access(config)
+  -- !!! Note thins function also executes fn_str !!!
+  -- If we support both old style (0.1.0) and new style (0.2.0+), this
+  -- early execution is mandatory, since it's our check for it.
+  local function load_function(fn_str)
+    local func1 = loadstring(fn_str)    -- load it
+    local _, func2 = pcall(func1)       -- run it
+    if type(func2) ~= "function" then
+      -- old style (0.1.0), without upvalues
+      return func1
+    else
+      -- this is a new function (0.2.0+), with upvalues
+      -- the first call to func1 above only initialized it, so run again
+      func2()
+      return func2
+    end
+  end
 
-    local functions = config_cache[config]
-
-    if not functions then
-      -- first call, go compile the functions
-      functions = {}
-      for _, fn_str in ipairs(config.functions) do
-        local func1 = loadstring(fn_str)    -- load it
-        local _, func2 = pcall(func1)       -- run it
-        if type(func2) ~= "function" then
-          -- old style (0.1.0), without upvalues
-          insert(functions, func1)
-        else
-          -- this is a new function (0.2.0+), with upvalues
-          insert(functions, func2)
-
-          -- the first call to func1 above only initialized it, so run again
-          func2()
-        end
+  local function invoke(phase, config)
+    if not config then
+      return
+    end
+
+    local cache = config_cache[config] or {}
+
+    -- (0.3.1) config.functions apply to access phase only
+    if phase == "access" and #config.functions > 0 and #config.access == 0 then
+      config.access = config.functions
```

Comment:
this is not ok. By altering `config` you're altering the config table that is a cached entity, which is owned by `ml-cache`.